fred2/campaigntreewnd.cpp

```diff
@@ -222,8 +222,8 @@ void campaign_tree_wnd::OnCpgnFileSaveAs()
 			return;		
 		}
 
-		string_copy(Campaign.filename, name, MAX_FILENAME_LEN);
-		string_copy(campaign_path, dlg.GetPathName(), 256);
+		string_copy(Campaign.filename, name, MAX_FILENAME_LEN - 1);
+		string_copy(campaign_path, dlg.GetPathName(), 256 - 1);
```

Comment:
How about `sizeof()` instead of 256?

Reply:
Eh, possible, but technically if `char` is ever more than 1 byte it wouldn't be correct